micrometer-core/src/main/java/io/micrometer/core/instrument/binder/okhttp3/OkHttpMetricsEventListener.java

```diff
@@ -100,15 +101,20 @@ public void responseHeadersEnd(Call call, Response response) {
     }
 
     private void time(CallState state) {
+        Request request = state.request;
+        boolean requestAvailable = request != null;
+
         String uri = state.response == null ? "UNKNOWN" :
-            (state.response.code() == 404 || state.response.code() == 301 ? "NOT_FOUND" : urlMapper.apply(state.request));
+            (state.response.code() == 404 || state.response.code() == 301 ? "NOT_FOUND" : urlMapper.apply(request));
+
+        Tags dynamicTags = requestAvailable ? request.tag(Tags.class) : Tags.empty();
```

Comment:
Out of curiosity, what happens if a user places a tag on the request that is not of type `Tags`?

Reply:
Then `request.tag(Tags.class)` returns `null`. `Request` holds a map as tag which typed as `<Class<T>, Object>` whose object is expected to have type `T`:
https://github.com/square/okhttp/blob/master/okhttp/src/main/java/okhttp3/Request.java#L94

Reply:
@jkschneider do you think there should be another marker class?

Reply:
Perfect. Thanks!